yjit/src/core.rs

```diff
@@ -126,6 +126,60 @@ impl Type {
         }
     }
 
+    /// Returns an Option with the T_ value type if it is known, otherwise None
+    pub fn known_value_type(&self) -> Option<ruby_value_type> {
+        match self {
+            Type::Nil => Some(RUBY_T_NIL),
+            Type::True => Some(RUBY_T_TRUE),
+            Type::False => Some(RUBY_T_FALSE),
+            Type::Fixnum => Some(RUBY_T_FIXNUM),
+            Type::Flonum => Some(RUBY_T_FLOAT),
+            Type::Array => Some(RUBY_T_ARRAY),
+            Type::Hash => Some(RUBY_T_HASH),
+            Type::ImmSymbol | Type::HeapSymbol => Some(RUBY_T_SYMBOL),
+            Type::TString | Type::CString => Some(RUBY_T_STRING),
+            Type::Unknown | Type::UnknownImm | Type::UnknownHeap => None
+        }
+    }
+
+    /// Returns an Option with the class if it is known, otherwise None
+    pub fn known_class_of(&self) -> Option<VALUE> {
```

Comment:
I think I would just rename this to `known_class`, which would be consistent with `known_exact_value` the "class_of" makes me think "is it a subclass of some other thing?".

Reply:
👍 `known_class` sounds better to me too. I also wondered if we should use `known_klass` as that's the spelling we used in `codegen.rs`, but I do prefer `known_class` 😓.